CommonCompilerConfig.cmake

```diff
@@ -159,16 +183,24 @@ endif()
 # try compiling without them.
 #
 if(NOT ${CMAKE_CXX_COMPILER_ID} STREQUAL "MSVC")
-  set(optimization_flags_cc "${optimization_flags_cc} -mtune=generic -O2")
+  set(optimization_flags_cc "${optimization_flags_cc} -O2")
   set(optimization_flags_cc "-pipe ${optimization_flags_cc}") #TODO use -Ofast instead of -O3
   set(optimization_flags_lt "-O2 ${optimization_flags_lt}")
 
+  CHECK_CXX_COMPILER_FLAG(-mtune=generic COMPILER_SUPPORTS_GENERIC_TUNING)
```

Comment:
I agree with @breznak's feedback regarding a comment. I would have expected this to be a check of `CMAKE_SYSTEM_PROCESSOR` for an ARM target instead of an arbitrary compiler flags test, which isn't self-documenting that we're actually dealing with ARM flags here (and elsewhere, too?).